src/app/components/serve/serve.component.ts

```diff
@@ -19,17 +20,26 @@ export class ServeComponent {
   command: AngularCliCommand;
   options = new NgserveOptions();
 
-  constructor(private commandService: CommandService) {}
+  constructor(
+    private commandService: CommandService,
+    private errorService: ErrorService
+    ) {}
 
   startServing(): void {
     this.sendCommand.emit(this.options.createCommandString());
   }
 
   stopServing(): void {
     this.commandService.stopServing(this.serveCommandId)
-    .subscribe(() => {
-      this.serveCommandId = null;
+    .subscribe(() => {},
+    error => {
+      this.errorService.addError({
+        errorText: 'The "ng serve" command you\'re trying to stop was not found',
+        errorDescription: 'The server is offline or have been restarted since you\'ve run this command'
+      });
     });
+    this.serveCommandId = null;
```

Comment:
should only be null if the stop command succeeds or fails, so try to add `finally`